frontend/src/pages/CompareV2.tsx

```diff
@@ -73,12 +94,29 @@ enum MetricsType {
   MARKDOWN,
 }
 
+const metricsTypeToString = (metricsType: MetricsType): string => {
+  switch (metricsType) {
+    case MetricsType.SCALAR_METRICS:
+      return 'Scalar Metrics';
+    case MetricsType.CONFUSION_MATRIX:
+      return 'Confusion Matrix';
+    case MetricsType.ROC_CURVE:
+      return 'ROC Curve';
+    case MetricsType.HTML:
+      return 'HTML';
+    case MetricsType.MARKDOWN:
+      return 'Markdown';
+    default:
+      return '';
+  }
+};
+
 // Include only the runs and executions which have artifacts of the specified type.
 function filterRunArtifactsByType(
-  runArtifacts: RunArtifacts[],
+  runArtifacts: RunArtifact[],
   artifactTypes: ArtifactType[],
   metricsType: MetricsType,
-): RunArtifacts[] {
+): RunArtifact[] {
   const metricsFilter =
```

Comment:
I would suggest to replace this part of logic with switch syntax. Which is more readable and maintainable.

Reply:
That makes sense - updated here: https://github.com/kubeflow/pipelines/pull/7943/commits/97222f137284ea0e032f574f8df8bccf3ca03fbc.